amida-tech/greyscale
# Task/inba 39 replace backend auth

#### What's this PR do?
Replaces the token auth strategy with a JwtStrategy for auth
Adds test cases for auth.

To run test, from `/backend` run:
`node_modules/.bin/mocha test/auth.integration.js` 

#### Related JIRA tickets:
[INBA-39](https://jira.amida-tech.com/browse/INBA-39)
#### How should this be manually tested?
Best way to test is with an HTTP client like postman.
- cp .env.example .env
- Make a GET request to: http://localhost:3005/public/v0.2/users/token
      - In postman select 'Authorization'
      - select type 'Basic Auth'
      - In "Username" field, enter "su@mail.net"
      - In "Password" field, enter "testuser"
      - Hit Send. You should get an output that looks something like:
   
<img width="872" alt="screen shot 2017-08-07 at 2 02 12 am" src="https://user-images.githubusercontent.com/1395745/29013964-823f6fce-7b14-11e7-9be7-3c3a20a54147.png">


- Using the token, make a request to a protected route. 
   http://localhost:3005/public/v0.2/organizations. Under "Headers", enter "Authorization" as the key and JWT + token as the value. And under "Body" enter "name": "Test organization" and "realm": "testorg"

<img width="873" alt="screen shot 2017-08-07 at 2 03 18 am" src="https://user-images.githubusercontent.com/1395745/29013989-a35589be-7b14-11e7-9478-f4315b8aff9f.png">

- Enter a an incorrect token to verify that you get an unauthorized error. 
- Test this with other protected routes.
#### Any background context you want to provide?
You'll need to have run the database scripts in backend/db_setup to populate the db with some data in order to be able to do this. See this [set up](https://paper.dropbox.com/doc/Indaba-Set-up-for-new-developers-Mac-5NzKPSGxbHOUXomHFaEKy) doc for more information.
#### Screenshots (if appropriate):
